src/storage/store/chunked_node_group.cpp

```diff
@@ -461,5 +464,36 @@ std::unique_ptr<ChunkedNodeGroup> ChunkedNodeGroup::deserialize(MemoryManager& m
     return chunkedGroup;
 }
 
+void ChunkedNodeGroup::setUnused(MemoryManager& mm) {
+    dataInUse = false;
```

Comment:
Should this be protected with lock too?

Reply:
I don't think it needs to be. The point of the lock is that once it gets added to the set of unused chunks any thread may attempt to spill it to disk, but before that point it should only be accessed by a single thread.